tests/rptest/services/redpanda.py

```diff
@@ -1819,20 +1819,29 @@ def save_executable(self):
             self._saved_executable = True
             self._context.log_collect['executable', self] = True
 
-    def search_log(self, pattern):
+    def search_log(self, pattern: str, node_matches: int = 1):
         """
         Test helper for grepping the redpanda log
 
+        :param pattern: the string to search for
+        :param node_matches: number of nodes that need a match before return True
+
         :return:  true if any instances of `pattern` found
         """
+        matches = [False] * len(self.nodes)
+
         for node in self.nodes:
             for line in node.account.ssh_capture(
                     f"grep \"{pattern}\" {RedpandaService.STDOUT_STDERR_CAPTURE} || true"
             ):
                 # We got a match
                 self.logger.debug(
                     f"Found {pattern} on node {node.name}: {line}")
-                return True
+                matches[self.idx(node) - 1] = True
```

Comment:
Why is an array of booleans needed here? It looks like this can be done with a counter that you increment each time you would have returned True.